### PR TITLE
Fixes to example ipynb files I found while testing examples

### PR DESCRIPTION
**Pull request by jspecht3**
**I went through all the example notebooks and corrected the errors I could find. I mostly ended up changing legacy functions and classes to the current/future supported functions.**

I changed the outdated functions and classes to those with future support. 
(e.g. changed function rectangular_prism to class RectangularPrism & changed class Source to class IndependentSource)

I changed the format for inputs that worked with old functions to inputs that work with current functions.
(e.g. The function openmc.plot_xs requires the reactions to be given as type dict, but was not before.)

I changed the initialization of the class EnergyGroups as EnergyGroups cannot be created without an initialization of group_edges.

**I did this as a means to learn GitHub, so I apologize if my descriptions are not apt enough or if I did not write with the requisite level of formality. However, I do believe it would be beneficial if these errors/out-of-date-code were to be corrected. Any feedback would be greatly appreciated!**

### CODE REVIEW

**Comment by pshriwise** [comment]
Thanks a ton for this @jspecht3! I've been meaning to get to this for some time now, but have been busy with many other things. I'll have a look as soon as I can.

**Comment by jspecht3** [comment]
Glad to hear that and I am happy to have helped!

**Comment by wahidluthfi** [comment]
Sorry for asking this silly question, but as of today, I can't open any example notebook from this git repo from my browser as I usually did before (last Friday). Also, this pull request is not updated in the notebook from the openmc documentation ([[https://github.com/openmc-dev/openmc/wiki/Example-Jupyter-Notebooks)](https://github.com/openmc-dev/openmc/wiki/Example-Jupyter-Notebooks))
Is there something on the administrator side that can solve this minor issue?

Here I am sending you some screenshots from my side
Screenshot_2024-05-20-07-02-20-862_com android chrome
Screenshot_2024-05-20-07-10-04-966_com android chrome

Note: I am reading the example notebook from my phone browser since it is a lot easier than opening my notebook on the bus. So I think the notebook itself is good but the preview function from Git Hub and the auto-update function from the main openmc documentation are affecting this issue.
Thanks

**Comment by pshriwise** [comment]
Hi @wahidluthfi, you're right about the preview of that notebook in the github repo. This is usually because github's preview software stack has moved beyond what the current version of the notebook are compatible with.

As for the preview of the notebooks in the OpenMC wiki -- those links point to the current version of the notebooks in the `main` branch of this repository. They will be updated once this PR has been merged in.

**Comment by jspecht3** [comment]
Hello, @wahidluthfi wahidluthfi, I am not sure about the previewer, but it appears that the OpenMC documentation you mentioned not being updated references the files edited in this pull request. I could be wrong about this, but it seems if this request is accepted, the documentation will automatically update as it specifically references these files.

image

**Comment by wahidluthfi** [comment]
Okay, thank you for the confirmation Patrick

File: mdgxs-part-ii.ipynb
Comment (pshriwise) [inline mdgxs-part-ii.ipynb:379]:
```suggestion
    "energy_groups = openmc.mgxs.EnergyGroups(group_edges=np.logspace(-3, 7.3, 21))\n",
```

File: mdgxs-part-ii.ipynb
Comment (pshriwise) [inline mdgxs-part-ii.ipynb:382]:
```suggestion
    "one_group = openmc.mgxs.EnergyGroups(group_edges=np.array([energy_groups.group_edges[0], energy_groups.group_edges[-1]]))"
```

File: mdgxs-part-i.ipynb
Comment (pshriwise) [inline mdgxs-part-i.ipynb:283]:
```suggestion
    "one_group = mgxs.EnergyGroups(group_edges=np.array([energy_groups.group_edges[0], energy_groups.group_edges[-1]]))\n",
```

File: mg-mode-part-iii.ipynb
Comment (pshriwise) [inline mg-mode-part-iii.ipynb:469]:
```suggestion
    "groups = openmc.mgxs.EnergyGroups(group_edges=np.array([0., 0.625, 20.0e6]))"
```

File: mgxs-part-i.ipynb
Comment (pshriwise) [inline mgxs-part-i.ipynb:321]:
```suggestion
    "groups = mgxs.EnergyGroups(group_edges=np.array([0., 0.625, 20.0e6]))\n"
```

File: mdgxs-part-i.ipynb
Comment (pshriwise) [inline mdgxs-part-i.ipynb:280]:
```suggestion
    "energy_groups = mgxs.EnergyGroups(group_edges=np.logspace(-3, 7.3, 101))\n",
```

File: pincell.ipynb
Comment (pshriwise) [inline pincell.ipynb:895]:
```suggestion
    "box = openmc.model.RectangularPrism(width=pitch, height=pitch,\n",
    "                                    boundary_type='reflective')\n",
```

**Review by pshriwise — CHANGES_REQUESTED** [review]
Thanks again for these updates @jspecht3! Mainly line comments from me and then we'll see how the testing goes.

**Comment by jspecht3** [comment]
@pshriwise I am glad to have been able to help! Please let me know how the testing goes.

**Comment by wahidluthfi** [comment]
Just a small update, now I can read the Jupyter Notebook from GitHub preview features without error. Regarding the repo, I think the current chain_simple.xml in this repo was missing some info regarding the Cs-136 because the depletion example didn't work. So I am using my old chain_simple downloaded a long time ago. I think there are also some pull requests regarding this issue.

**Comment by pshriwise** [comment]
> Many thanks for keeping these upto date
> 
> We could perhaps think about adding some CI on the openmc repo that runs these notebooks whenever there is a PR on openmc repo from develop to master.
> 
> This could help ensure the notebooks match the latest stable release of openmc
> 
> I'm happy to make the proposed CI if anyone thinks it is a good idea

by all means!

**Comment by pshriwise** [comment]
Sorry for pushing to your branch so much @jspecht3! There were some additional changes needed to update the CI runner to v0.14.0 and some changes needed in the sample chain file.

**Comment by jspecht3** [comment]
@pshriwise Please push as many changes as you need and do not hesitate to let me know if there is something I need to do.

**Review by pshriwise — APPROVED** [review]
The notebooks run well with 0.14.0 now @jspecht3! Sorry again for all of the changes pushed to your branch. That's not normally how we do this, but there were a lot of changes needed to the CI environment.

I ended up removing testing for the part i and ii multi-group cross section tests for now. They require a change in OpenMOC to be pulled into the CI environment here. Once I've verified that this fix is working I'll submit another PR here to start testing those again:

https://github.com/mit-crpg/OpenMOC/pull/484

So, with your approval @jspecht3, I'd like to merge this to get us up-to-date so we can continue development cleanly from there.

**Comment by jspecht3** [comment]
@pshriwise You have my approval to merge this branch. Thank you for the experience, this whole process has been very enjoyable! I hope to be able to help in the future.

**Pull request merged**
